### DISCUSSION TITLE
[Bug]:TRESTDWIdDatabase do Delphi 10.4 community não reconhece o pooler do servidor

### DISCUSSION THREAD

**Discussion by ClaudemirLopes** [OpenSourceCommunityBrasil/REST-DataWare#232] [General]
### Descrição do bug

Pessoal, estou realizando alguns testes no Delphi para uma aplicação em 3 camadas. Para tal, criei um servidor de aplicação utilizando o TRESTDWServiceProoler que está iniciando e passando pelo teste localhos:8082, por exemplo. No entanto, quando tento localizar esse Pooler no Client, o mesmo não aparece. Alguém já teve essa diculdade? Se sim, poderiam me ajudar a solucioná-la?

Obrigado

Claudemir Lopes

### Passos para reproduzir o problema

Criei um servidor  TRESTDWServiceProoler em StandAlone e tente conectar o cliente com o TRESTDEIdDatabase utilizando a propriedade PoolerName. Se o TRESTDWServicePooler não aparecer, eis o problema.

### Versão RDW:

2.1.0 (branch)

### IDE:

Delphi novo

### Sistema Operacional:

Windows

### Relator do Problema

ClaudemirLopes

**Comment by FernandoBanhos** [comment]
abre o server em modo release, sem ser por debug...

**Comment by ClaudemirLopes** [comment]
Estou abrindo diretamente no executável e mesmo assim, não funciona.

**Comment by FernandoBanhos** [comment]
tira um print do codigo fonte do seu datamodule

**Comment by ClaudemirLopes** [comment]
unit uDmPrincipal;

interface

uses
  System.SysUtils, System.Classes, uRESTDWDatamodule, uRESTDWComponentBase,
  uRESTDWServerEvents, uRESTDWConsts, uRESTDWParams, System.JSON,
  uRESTDWBasicDB, uRESTDWIdBase, FireDAC.Stan.Intf, FireDAC.Stan.Option,
  FireDAC.Stan.Error, FireDAC.UI.Intf, FireDAC.Phys.Intf, FireDAC.Stan.Def,
  FireDAC.Stan.Pool, FireDAC.Stan.Async, FireDAC.Phys, FireDAC.Phys.PG,
  FireDAC.Phys.PGDef, FireDAC.VCLUI.Wait, Data.DB, FireDAC.Comp.Client,
  uRESTDWBasic, uRestDWDriverFD, FireDAC.Stan.Param, FireDAC.DatS,
  FireDAC.DApt.Intf, FireDAC.DApt, FireDAC.Comp.DataSet, FireDAC.Comp.UI;

type
  TdmPrincipal = class(TServerMethodDataModule)
    DWDriveFd: TRESTDWDriverFD;
    FdDB: TFDConnection;
    FdPgDrive: TFDPhysPgDriverLink;
    qryClientes: TFDQuery;
    qryClientesnome: TWideStringField;
    qryClientesativo: TWideStringField;
    qryClientesid: TLargeintField;
    FDGUIxWaitCursor1: TFDGUIxWaitCursor;
  private
    { Private declarations }

    { Public declarations }
  end;

var
  dmPrincipal: TdmPrincipal;

implementation

{%CLASSGROUP 'Vcl.Controls.TControl'}

uses uFrmPrincipal;

{$R *.dfm}



end.

**Comment by gledstonprego** [comment]
suba o server e como cliente aberto no delphi RESTDWIdDatabase1 va na propriedade poolername
e veja se aparece ali so com o server ativo

**Comment by ClaudemirLopes** [comment]
Então, eu já fiz isso e não aparece. O estranho é que se eu for no browser e digitar localhost:8082, o mesmo me retorna que o server está on-line. Inclusive, piguei o meu ip em outra máquina e o server respondeu que está on-line também. O server eu tenho a certeza que está funcionando!

**Comment by mobius1qwe** [comment]
Faltou o PoolerDB.

**Comment by ClaudemirLopes** [comment]
Infelizmente, mesmo eu inserindo o RESTDWPoolerDB e vinculando o RESTDriver no dmPrincipal.DWDriveFd, não funcionou.

**Comment by FernandoBanhos** [comment]
me chama no PV, se nao conseguir...

**Comment by gledstonprego** [comment]
pelo visto voce abriu o demo e com 2 DM pode ser que voce esta definindo um DM e apontando para outro
verifique se esta apontando para o DM que voce esta informando com ativo
